src/qml/editorwidgets/ExternalResource.qml

```diff
@@ -16,43 +17,80 @@ Item {
 
   property PictureSource __pictureSource
 
-  Image {
-    property var currentValue: value
+  //on all mimetypes image/... and on empty values it should appear as an image widget
+  property bool isImage: FileUtils.mimeTypeName( qgisProject.homePath + '/' + value ).startsWith("image/") || FileUtils.fileName( qgisProject.homePath + '/' + value ) === ''
+
+  Label {
+    id: linkField
+    height: fontMetrics.height + 20 * dp
+    topPadding: 10 * dp
+    bottomPadding: 10 * dp
+    visible: !isImage
+    anchors.left: parent.left
+    anchors.right: parent.right
+    font: Theme.defaultFont
+    color: FileUtils.fileExists(qgisProject.homePath + '/' + value) ? '#0000EE' : 'black'
+
+    text: FileUtils.fileName( qgisProject.homePath + '/' + value )
+
+    background: Rectangle {
+      y: linkField.height - height - linkField.bottomPadding / 2
+      implicitWidth: 120 * dp
+      height: 1 * dp
+      color: "#C8E6C9"
+    }
+
+    MouseArea {
+      anchors.fill: parent
+
+      onClicked: {
+        if (value && FileUtils.fileExists(qgisProject.homePath + '/' + value) )
+          platformUtilities.open( qgisProject.homePath + '/' + value );
+      }
+    }
+  }
 
+  FontMetrics {
+    id: fontMetrics
+    font: linkField.font
+  }
+
+  Image {
     id: image
+    visible: isImage
     width: 200 * dp
     autoTransform: true
     fillMode: Image.PreserveAspectFit
     horizontalAlignment: Image.AlignLeft
 
-    //source is managed over onCurrentValueChanged since the binding would break somewhere
-    source: Theme.getThemeIcon("ic_photo_notavailable_white_48dp")
+    source: if ( FileUtils.fileName( qgisProject.homePath + '/' + value ) !== '' ) {
+                if( FileUtils.fileExists(qgisProject.homePath + '/' + value) && image.status !== Image.Error ) {
+                    'file://' + qgisProject.homePath + '/' + value
+                } else {
+                  Theme.getThemeIcon("ic_broken_image_black_24dp")
```

Comment:
Shouldn't we have all the same sizes of (48dp) of the images?

Reply:
Yes. I think it's the name that is confusing. The images are the same size depending on the dpi, if we have only one of them.
What should we do? Rename them all?

Reply:
I am not sure I understand your statement, are they the same (pixel) resolution or not?

Reply:
Yes. Same Pixel Resolution

Reply:
Then they should get the same dp suffix.

Reply:
On MDPI they are 24px...
This means `ic_photo_notavailable_white_48dp` should be renamed to  `ic_photo_notavailable_black_24dp`, okay?

The size is not defined in QML so it's `24 * dp` I guess. Or do I misunderstand still something?